onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -207,23 +207,77 @@ MlasGemm(
     MLAS_THREADPOOL* ThreadPool
     );
 
-void
-MLASCALL
-MlasGemm(
-    size_t M,
-    size_t N,
-    size_t K,
-    const uint8_t* A,
-    size_t lda,
-    uint8_t offa,
-    const uint8_t* B,
-    size_t ldb,
-    uint8_t offb,
-    bool BIsSigned,
-    int32_t* C,
-    size_t ldc,
-    MLAS_THREADPOOL* ThreadPool
-    );
+enum class MLAS_QUANTIZATION_GRANULARITY {
+    PerMatrix,
+    PerColumn,
+};
+
+enum class MLAS_QGEMM_OUTPUT_MODE {
+    ZeroMode,       // overwrite the output buffer
+    AccumulateMode, // accumulate to the output buffer
+};
+
+class MLAS_QGEMM_OUTPUT_PROCESSOR {
+public:
+    inline
+    virtual
+    void
+    Process(
+        const int32_t* /*CBuffer*/,
```

Comment:
I would prefer the existing MLAS style of declaring the local and then using MLAS_UNREFERENCED_PARAMETER for the unused parameters.

Reply:
MLAS_UNREFERENCED_PARAMETER is in the mlasi.h. I changed this function to pure virtual function.

---
In reply to: [518283288](https://github.com/microsoft/onnxruntime/pull/5632#discussion_r518283288) [](ancestors = 518283288)